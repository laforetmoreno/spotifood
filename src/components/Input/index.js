import React from 'react';
import { string, func } from 'prop-types';

import styles from './index.module.scss';

const Input = ({ name, onChange, value, placeholder, label }) => {
  const handleChange = e => {
    if (onChange) onChange(e.target.value);
  };

  return (
    <div className={styles.wrapper}>
      <span className={styles.label}>{label}</span>
      <input
        className={styles.input}
        data-testid="input"
        name={name}
        onChange={handleChange}
        value={value}
        placeholder={placeholder}
      />
    </div>
  );
};

Input.propTypes = {
  onChange: func,
  name: string,
  value: string,
  placeholder: string,
  label: string,
};

Input.defaultProps = {
  onChange: () => '',
  name: '',
  value: '',
  placeholder: '',
  label: string,
};

export default Input;
