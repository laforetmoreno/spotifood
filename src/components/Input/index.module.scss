@import '../../styles/';

.wrapper {
  @include filters-wrapper-default();
}

.input {
  font-size: rem(16px);
  outline: none;
  border: none;
  padding: rem(6px);
  border-radius: rem(4px);
  margin-top: rem(10px);
  background: $--color-green;

  &::placeholder {
    color: $--color-black;
    opacity: 0.4;
  }
}

.label {
  color: $--color-green;
}
