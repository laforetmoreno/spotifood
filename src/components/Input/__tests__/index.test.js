import React from 'react';
import { render, waitForElement, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Input from '../index';

describe('<Input />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a input with placeholder', () => {
    const tree = renderer.create(<Input placeholder="Exemple" />);
    expect(tree).toMatchSnapshot();
  });

  it('get by id and change', async () => {
    const { getByTestId } = render(<Input />);
    const fieldNode = await waitForElement(() => getByTestId('input'));
    fireEvent.change(fieldNode, { target: { value: 'test change' } });
  });

  it('get a placeholder', async () => {
    const { getByPlaceholderText } = render(<Input placeholder="fake placeholder" />);
    const fieldNode = await waitForElement(() => getByPlaceholderText('fake placeholder'));
    fireEvent.change(fieldNode);
  });
});
