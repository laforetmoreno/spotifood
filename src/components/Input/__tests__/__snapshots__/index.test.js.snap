// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
<div
  className="wrapper"
>
  <span
    className="label"
  />
  <input
    className="input"
    data-testid="input"
    name=""
    onChange={[Function]}
    placeholder=""
    value=""
  />
</div>
`;

exports[`<Input /> should render a input with placeholder 1`] = `
<div
  className="wrapper"
>
  <span
    className="label"
  />
  <input
    className="input"
    data-testid="input"
    name=""
    onChange={[Function]}
    placeholder="Exemple"
    value=""
  />
</div>
`;
